### PR TITLE
feat(debug): bug report export and repeat task in debug panel (ENG-693)

### LINKED ISSUE #404: [Feature] Improve Debug Mode Reporting
Improve debug mode to easily report bugs (include axtrees snapshot and screenshots). like a button that says ("repeat last tasks and create a bug report")

### PR DESCRIPTION
**Pull request by mavishay**
## Summary

Adds **Bug Report** and **Repeat Task** buttons to the debug panel (visible only when debug mode is enabled).

### 🐛 Bug Report
One click captures and saves a structured diagnostic file:
- **Screenshot** via Electron `capturePage()` (base64 PNG)
- **Accessibility tree** via `executeJavaScript` DOM serialization (with depth/node limits)
- **Task metadata**: prompt, status, timestamps, message count
- **Debug logs**: full log array
- Saved as JSON via native save dialog + optional PNG sidecar

### 🔁 Repeat Task
Re-runs the same prompt to reproduce issues.

## Changes

| File | From | Change |
|------|------|--------|
| `apps/desktop/src/main/ipc/handlers.ts` | #549 @Kesavaraja67 + #510 @aryan867 | `debug:capture-screenshot`, `debug:capture-axtree`, `debug:generate-bug-report` IPC handlers |
| `apps/desktop/src/preload/index.ts` | #549 @Kesavaraja67 + #510 @aryan867 | Preload bridge for all three handlers |
| `apps/web/src/client/lib/accomplish.ts` | #549 @Kesavaraja67 + #614 @SaaiAravindhRaja | `captureScreenshot()`, `captureAxtree()`, `generateBugReport()` API |
| `apps/web/src/client/components/execution/DebugPanel.tsx` | #614 @SaaiAravindhRaja | Bug Report + Repeat Task buttons |
| `apps/web/src/client/pages/Execution.tsx` | #614 @SaaiAravindhRaja | `handleBugReport` + `handleRepeatTask` handlers |
| `apps/desktop/__tests__/unit/main/ipc/handlers.unit.test.ts` | #510 @aryan867 | Unit tests for all three IPC handlers |
| `apps/desktop/__tests__/unit/main/opencode/auth-browser.unit.test.ts` | #549 @Kesavaraja67 | Mock path fix |

### @Kesavaraja67 (PR #549) — IPC Architecture
The complete bug report IPC handler concept with screenshot + axtree capture, structured JSON export, and the preload bridge. Also includes the auth-browser test mock path fix.

### @SaaiAravindhRaja (PR #614) — Debug Panel UI
The Bug Report and Repeat Task button integration in `DebugPanel.tsx` and `Execution.tsx` — clean prop-based design that fits the existing panel, uses `@phosphor-icons/react`.

### @aryan867 (PR #510) — Unit Tests + IPC Design
Comprehensive unit tests for all three debug IPC handlers covering screenshot capture, axtree serialization, file save dialog, cancellation, write errors, and no-window cases.

Closes #404

## Summary by CodeRabbit

* **New Features**
  * Add end-user bug reporting: capture screenshots, collect accessibility tree data, and generate/save consolidated bug reports (optional screenshot sidecar).
  * Expose debug actions in the UI and API: "Bug Report" and "Repeat Task" controls in the Debug Panel with status indicators; new client API methods to capture screenshot/axtree and generate reports.

* **Tests**
  * Add extensive tests covering debug handlers, screenshot/axtree capture, bug-report generation flows, cancellations, write errors, and edge cases.

### CODE REVIEW

**Comment by orcaman** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/or.hiltch/pr-checks/587627ee-8a4c-4949-9dcb-6c0a018176c0?refs=652a135d4fff99cff9e098bbde10935bbc497712&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/or.hiltch/pr-checks/587627ee-8a4c-4949-9dcb-6c0a018176c0/license?refs=652a135d4fff99cff9e098bbde10935bbc497712&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: apps/desktop/src/main/ipc/handlers.ts [hunk 1]
```diff
@@ -1447,6 +1447,147 @@ export function registerIPCHandlers(): void {
...
+          fs.writeFileSync(screenshotPath, Buffer.from(reportData.screenshot, 'base64'));
+        }
+
+        return { success: true, path: result.filePath };
+      } catch (error) {
+        const message = error instanceof Error ? error.message : 'Unknown error';
+        return { success: false, error: message };
+      }
+    },
+  );
```
[the remark this answers is not in the document]
Comment (mavishay) [reply apps/desktop/src/main/ipc/handlers.ts:1401]: Fixed — added `assertDebugModeEnabled()` helper that checks `storage.getDebugMode()` and throws if disabled. Called at the top of all 3 debug handlers (`capture-screenshot`, `capture-axtree`, `generate-bug-report`) before any privileged operations, returning `{ success: false, error }` on failure.

File: apps/desktop/src/main/ipc/handlers.ts [hunk 2]
```diff
@@ -1447,6 +1447,147 @@ export function registerIPCHandlers(): void {
...
+          hasScreenshot: Boolean(reportData.screenshot),
+        };
+
+        fs.writeFileSync(result.filePath, JSON.stringify(report, null, 2));
+
+        if (reportData.screenshot) {
+          const parsed = path.parse(result.filePath);
+          const screenshotPath = path.join(parsed.dir, `${parsed.name}.png`);
+          fs.writeFileSync(screenshotPath, Buffer.from(reportData.screenshot, 'base64'));
+        }
```
[the remark this answers is not in the document]
Comment (mavishay) [reply apps/desktop/src/main/ipc/handlers.ts]: Fixed — replaced both `fs.writeFileSync` calls with `await fs.promises.writeFile` so the Electron main thread is not blocked during bug report export.

File: apps/desktop/src/main/ipc/handlers.ts [hunk 3]
```diff
@@ -1447,6 +1447,171 @@ export function registerIPCHandlers(): void {
...
+            version: reportData.appVersion ?? app.getVersion(),
+            platform: reportData.platform ?? process.platform,
+          },
+          task: {
+            id: reportData.taskId,
+            prompt: reportData.taskPrompt,
+            status: reportData.taskStatus,
+          },
+          messages: reportData.messages,
+          debugLogs: reportData.debugLogs,
```
[the remark this answers is not in the document]
Comment (mavishay) [reply apps/desktop/src/main/ipc/handlers.ts:1387]: Fixed — added `taskCreatedAt`, `taskUpdatedAt`, `taskCompletedAt`, and `messageCount` to both the `reportData` type and the exported JSON. Timestamps are wired from `currentTask` in `Execution.tsx`.

File: apps/desktop/src/main/ipc/handlers.ts [hunk 4]
```diff
@@ -1447,6 +1447,171 @@ export function registerIPCHandlers(): void {
...
+          hasScreenshot: Boolean(reportData.screenshot),
+        };
+
+        await fs.promises.writeFile(result.filePath, JSON.stringify(report, null, 2), 'utf-8');
+
+        if (reportData.screenshot) {
+          const parsed = path.parse(result.filePath);
+          const screenshotPath = path.join(parsed.dir, `${parsed.name}.png`);
+          await fs.promises.writeFile(screenshotPath, Buffer.from(reportData.screenshot, 'base64'));
+        }
```
[the remark this answers is not in the document]
Comment (mavishay) [reply apps/desktop/src/main/ipc/handlers.ts]: Fixed — PNG sidecar now checks for existing file via `fs.promises.access` before writing. If the file exists, a timestamped fallback name is used instead of silently overwriting. `hasScreenshot` is set only after the write succeeds.

File: apps/desktop/src/main/ipc/handlers.ts [hunk 5]
```diff
@@ -1447,6 +1447,189 @@ export function registerIPCHandlers(): void {
...
+    try {
+      assertDebugModeEnabled();
+    } catch (error) {
+      const message = error instanceof Error ? error.message : 'Debug mode disabled';
+      return { success: false, error: message };
+    }
+    const window = BrowserWindow.fromWebContents(event.sender);
+    if (!window) {
+      return { success: false, error: 'No window found' };
+    }
```
[the remark this answers is not in the document]
Comment (mavishay) [reply apps/desktop/src/main/ipc/handlers.ts:1238]: Fixed — all 3 debug handlers now use assertTrustedWindow() with try/catch before any privileged operations, returning { success: false, error } on trust failure.

File: apps/desktop/src/main/ipc/handlers.ts [hunk 6]
```diff
@@ -1447,6 +1447,189 @@ export function registerIPCHandlers(): void {
...
+            createdAt: reportData.taskCreatedAt,
+            updatedAt: reportData.taskUpdatedAt,
+            completedAt: reportData.taskCompletedAt,
+            messageCount: Array.isArray(reportData.messages) ? reportData.messages.length : 0,
+          },
+          messages: reportData.messages,
+          debugLogs: reportData.debugLogs,
+          axtree: reportData.axtree,
+          screenshotPath: screenshotPath ?? null,
+          hasScreenshot: Boolean(screenshotPath),
```
[the remark this answers is not in the document]
Comment (mavishay) [reply apps/desktop/src/main/ipc/handlers.ts:1390]: Fixed — replaced the absolute screenshotPath with path.basename(screenshotPath) stored as screenshotFile in the JSON. The full local path is no longer included in the exported report.

File: apps/web/src/client/pages/Execution.tsx [hunk 7]
```diff
@@ -493,6 +496,54 @@ export function ExecutionPage() {
...
+    setRepeatingTask(true);
+    try {
+      const newTask = await accomplish.startTask({ prompt: currentTask.prompt });
+      navigate(`/execution/${newTask.id}`);
+    } catch (err) {
+      console.error('[Execution] Failed to repeat task:', err);
+    } finally {
+      setRepeatingTask(false);
+    }
+  }, [accomplish, currentTask, navigate]);
```
[the remark this answers is not in the document]
Comment (mavishay) [reply apps/web/src/client/pages/Execution.tsx:553]: Fixed — added an activeStatuses guard at the top of handleRepeatTask. Tasks in pending, queued, running, waiting_permission, or waiting states are blocked from being repeated.

File: apps/desktop/src/main/ipc/handlers.ts [hunk 8]
```diff
@@ -1447,6 +1447,198 @@ export function registerIPCHandlers(): void {
...
+        taskCreatedAt?: string;
+        taskUpdatedAt?: string;
+        taskCompletedAt?: string;
+        messages?: unknown[];
+        debugLogs?: unknown[];
+        screenshot?: string;
+        axtree?: string;
+        appVersion?: string;
+        platform?: string;
+      },
```
[the remark this answers is not in the document]
Comment (mavishay) [reply apps/desktop/src/main/ipc/handlers.ts:1324]: Fixed — added `taskCreatedAt`, `taskUpdatedAt`, and `taskCompletedAt` to the preload `generateBugReport` type definition, aligning it with the main handler and `accomplish.ts` interface.

**Comment by mavishay** [comment]
## ✅ Merged — Thank you all!

This PR brings a solid debug tooling feature to the app. Really appreciate the collaborative effort here:

**@Kesavaraja67** — The IPC architecture was the backbone of this whole feature. The screenshot + axtree capture handlers, structured JSON export, and preload bridge were all well-designed and clean.

**@SaaiAravindhRaja** — The Debug Panel UI integration was smooth and fit naturally into the existing component structure. The prop-based design for Bug Report + Repeat Task buttons was exactly the right approach.

**@aryan867** — Comprehensive unit tests covering all the edge cases (cancellations, write errors, no-window cases) — this is what gives us confidence to ship. Great work.

Closes #404 🎉
